From 78ce6963dc772ec17086c553cad91500686bd03a Mon Sep 17 00:00:00 2001
From: Chaoyi Chen <chaoyi.chen@rock-chips.com>
Date: Tue, 12 Dec 2023 17:02:58 +0800
Subject: [PATCH 197/232] misc: rk628: combtxphy: Modify to meet LVDS standard
 requirement

This patch Modify TX common output voltage to
meet LVDS standard requirements.

Change-Id: I78275ad02b556c80d23dc20b915603a3e2295289
Signed-off-by: Chaoyi Chen <chaoyi.chen@rock-chips.com>
Signed-off-by: Guochun Huang <hero.huang@rock-chips.com>
---
 drivers/misc/rk628/rk628_combtxphy.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/misc/rk628/rk628_combtxphy.c b/drivers/misc/rk628/rk628_combtxphy.c
index 9a885b4babdf5c..2d46b6ffb1833f 100644
--- a/drivers/misc/rk628/rk628_combtxphy.c
+++ b/drivers/misc/rk628/rk628_combtxphy.c
@@ -57,11 +57,15 @@ static void rk628_combtxphy_lvds_power_on(struct rk628 *rk628)
 	int ret;
 
 	/* Adjust terminal resistance 133 ohm, bypass 0.95v ldo for driver. */
+	if (rk628->version == RK628F_VERSION)
+		val = TX_COM_VOLT_ADJ(3);
+	else
+		val = TX_COM_VOLT_ADJ(0);
 	rk628_i2c_update_bits(rk628, COMBTXPHY_CON7,
 			      SW_TX_RTERM_MASK | SW_TX_MODE_MASK |
 			      BYPASS_095V_LDO_MASK | TX_COM_VOLT_ADJ_MASK,
 			      SW_TX_RTERM(6) | SW_TX_MODE(3) |
-			      BYPASS_095V_LDO(1) | TX_COM_VOLT_ADJ(0));
+			      BYPASS_095V_LDO(1) | val);
 
 	rk628_i2c_write(rk628, COMBTXPHY_CON10, TX7_CKDRV_EN | TX2_CKDRV_EN);
 	rk628_i2c_update_bits(rk628, COMBTXPHY_CON0,
-- 
2.17.1

